File: backup_java/dialogs/DialogMerchant.java
package it.paranoidsquirrels.idleguildmaster.ui.dialogs;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ListAdapter;
import androidx.viewbinding.ViewBinding;
import it.paranoidsquirrels.idleguildmaster.MainActivity;
import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.TrueTimeUtils;
import it.paranoidsquirrels.idleguildmaster.UIUtils;
import it.paranoidsquirrels.idleguildmaster.Utils;
import it.paranoidsquirrels.idleguildmaster.databinding.DialogMerchantBinding;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.MerchantOffer;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Upgrade;
import java.util.ArrayList;
import java.util.Iterator;
import java.util.function.BooleanSupplier;

/* JADX INFO: loaded from: classes3.dex */
public class DialogMerchant extends CustomDialog {
    public DialogMerchantBinding binding;

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected ViewBinding getBinding() {
        return this.binding;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void setBinding(ViewBinding viewBinding) {
        this.binding = (DialogMerchantBinding) viewBinding;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected String getTitle() {
        return getString(R.string.merchant_dialog_title);
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected ViewBinding inflate(LayoutInflater layoutInflater, ViewGroup viewGroup, boolean z) {
        DialogMerchantBinding dialogMerchantBindingInflate = DialogMerchantBinding.inflate(layoutInflater, viewGroup, z);
        this.binding = dialogMerchantBindingInflate;
        return dialogMerchantBindingInflate;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void initialize(Bundle bundle) {
        this.binding.newRegular.setVisibility(MainActivity.data.isNewMerchantRegularItems() ? 0 : 4);
        this.binding.newSpecial.setVisibility(MainActivity.data.isNewMerchantSpecialItems() ? 0 : 4);
        MainActivity.data.setNewMerchantRegularItems(false);
        MainActivity.data.setNewMerchantSpecialItems(false);
        ((MainActivity) getActivity()).refreshIcons();
        ArrayList arrayList = new ArrayList();
        Iterator<MerchantOffer> it2 = MainActivity.data.getMerchantRegularStockItems().iterator();
        while (it2.hasNext()) {
            arrayList.add(it2.next().getItem());
        }
        this.binding.regularItemGrid.setAdapter((ListAdapter) UIUtils.getItemsGridAdapter(getContext(), arrayList));
        ArrayList arrayList2 = new ArrayList();
        Iterator<MerchantOffer> it3 = MainActivity.data.getMerchantSpecialReserve().iterator();
        while (it3.hasNext()) {
            arrayList2.add(it3.next().getItem());
        }
        this.binding.specialItemGrid.setAdapter((ListAdapter) UIUtils.getItemsGridAdapter(getContext(), arrayList2));
        boolean zIsEmpty = MainActivity.data.getMerchantRegularStockItems().isEmpty();
        this.binding.regularItemGrid.setVisibility(zIsEmpty ? 8 : 0);
        this.binding.noRegularItems.setVisibility(zIsEmpty ? 0 : 8);
        boolean zIsEmpty2 = MainActivity.data.getMerchantSpecialReserve().isEmpty();
        this.binding.specialItemGrid.setVisibility(zIsEmpty2 ? 8 : 0);
        this.binding.noSpecialItems.setVisibility(zIsEmpty2 ? 0 : 8);
    }

    @Override // androidx.fragment.app.Fragment
    public void onResume() {
        super.onResume();
        Utils.refreshCooldowns(TrueTimeUtils.millis());
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void attachListeners() {
        this.binding.regularItemGrid.setOnItemClickListener(new AdapterView.OnItemClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogMerchant$$ExternalSyntheticLambda1
            @Override // android.widget.AdapterView.OnItemClickListener
            public final void onItemClick(AdapterView adapterView, View view, int i, long j) {
                DialogMerchant.this.m357x3436fa29(adapterView, view, i, j);
            }
        });
        this.binding.specialItemGrid.setOnItemClickListener(new AdapterView.OnItemClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogMerchant$$ExternalSyntheticLambda2
            @Override // android.widget.AdapterView.OnItemClickListener
            public final void onItemClick(AdapterView adapterView, View view, int i, long j) {
                DialogMerchant.this.m358xb297fe08(adapterView, view, i, j);
            }
        });
        this.binding.regularItemGrid.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogMerchant$$ExternalSyntheticLambda3
            @Override // android.widget.AdapterView.OnItemLongClickListener
            public final boolean onItemLongClick(AdapterView adapterView, View view, int i, long j) {
                return DialogMerchant.this.m359x30f901e7(adapterView, view, i, j);
            }
        });
        this.binding.specialItemGrid.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogMerchant$$ExternalSyntheticLambda4
            @Override // android.widget.AdapterView.OnItemLongClickListener
            public final boolean onItemLongClick(AdapterView adapterView, View view, int i, long j) {
                return DialogMerchant.this.m360xaf5a05c6(adapterView, view, i, j);
            }
        });
        this.binding.close.setOnClickListener(new View.OnClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogMerchant$$ExternalSyntheticLambda5
            @Override // android.view.View.OnClickListener
            public final void onClick(View view) {
                DialogMerchant.this.m361x2dbb09a5(view);
            }
        });
    }

    /* JADX INFO: renamed from: lambda$attachListeners$0$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogMerchant, reason: not valid java name */
    /* synthetic */ void m357x3436fa29(AdapterView adapterView, View view, int i, long j) {
        openBuyDialog(MainActivity.data.getMerchantRegularStockItems().get(i), false);
    }

    /* JADX INFO: renamed from: lambda$attachListeners$1$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogMerchant, reason: not valid java name */
    /* synthetic */ void m358xb297fe08(AdapterView adapterView, View view, int i, long j) {
        openBuyDialog(MainActivity.data.getMerchantSpecialReserve().get(i), true);
    }

    /* JADX INFO: renamed from: lambda$attachListeners$2$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogMerchant, reason: not valid java name */
    /* synthetic */ boolean m359x30f901e7(AdapterView adapterView, View view, int i, long j) {
        UIUtils.vibrate(getContext());
        UIUtils.openItemDetail(MainActivity.data.getMerchantRegularStockItems().get(i).getItem());
        return true;
    }

    /* JADX INFO: renamed from: lambda$attachListeners$3$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogMerchant, reason: not valid java name */
    /* synthetic */ boolean m360xaf5a05c6(AdapterView adapterView, View view, int i, long j) {
        UIUtils.vibrate(getContext());
        UIUtils.openItemDetail(MainActivity.data.getMerchantSpecialReserve().get(i).getItem());
        return true;
    }

    /* JADX INFO: renamed from: lambda$attachListeners$4$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogMerchant, reason: not valid java name */
    /* synthetic */ void m361x2dbb09a5(View view) {
        dismiss();
    }

    private void openBuyDialog(final MerchantOffer merchantOffer, final boolean z) {
        if (MainActivity.shownDialogBuyFromMerchant != null) {
            return;
        }
        DialogBuyFromMerchant dialogBuyFromMerchant = new DialogBuyFromMerchant();
        dialogBuyFromMerchant.offer = merchantOffer;
        dialogBuyFromMerchant.callback = new BooleanSupplier() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogMerchant$$ExternalSyntheticLambda0
            @Override // java.util.function.BooleanSupplier
            public final boolean getAsBoolean() {
                return DialogMerchant.this.m362x39fcb3a(merchantOffer, z);
            }
        };
        dialogBuyFromMerchant.show(getParentFragmentManager(), "select_equipment");
    }

    /* JADX INFO: renamed from: lambda$openBuyDialog$5$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogMerchant, reason: not valid java name */
    /* synthetic */ boolean m362x39fcb3a(MerchantOffer merchantOffer, boolean z) {
        if (!(merchantOffer.getItem() instanceof Upgrade) && Utils.remainingInventorySpaceAfterCollecting(false, merchantOffer.getItem()) < 0) {
            MainActivity.shownDialogBuyFromMerchant.writeError(getString(R.string.error_not_enough_space));
            return false;
        }
        if (merchantOffer.isGems()) {
            long gems = MainActivity.data.getGems() - merchantOffer.getPrice();
            if (gems < 0) {
                MainActivity.shownDialogBuyFromMerchant.writeError(getString(R.string.error_not_enough_gems));
                return false;
            }
            MainActivity.data.setGems(gems);
            ((MainActivity) getActivity()).refreshGems();
        } else {
            long money = MainActivity.data.getMoney() - merchantOffer.getPrice();
            if (money < 0) {
                MainActivity.shownDialogBuyFromMerchant.writeError(getString(R.string.error_not_enough_money));
                return false;
            }
            MainActivity.data.setMoney(money);
            ((MainActivity) getActivity()).refreshMoney();
        }
        if (z) {
            MainActivity.data.getMerchantSpecialReserve().remove(merchantOffer);
        } else {
            MainActivity.data.getMerchantRegularStockItems().remove(merchantOffer);
        }
        if (merchantOffer.getItem().getUniqueOrigin() != null) {
            MainActivity.data.getUniqueItemsLost().remove(merchantOffer.getItem().getUniqueOrigin());
        }
        if (merchantOffer.getItem() instanceof Upgrade) {
            ((Upgrade) merchantOffer.getItem()).use();
        } else {
            Utils.collectItem(merchantOffer.getItem(), MainActivity.data.getItems());
        }
        MainActivity.headquartersFragment.refresh();
        initialize(null);
        return true;
    }

    public void refreshCooldowns(int i, int i2, int i3) {
        this.binding.regularItemsCountdown.setText(String.format(getString(R.string.time_hours_minutes), Integer.valueOf(i2), Integer.valueOf(i3)));
        this.binding.specialItemsCountdown.setText(String.format(getString(R.string.time_days_hours_minutes), Integer.valueOf(i), Integer.valueOf(i2), Integer.valueOf(i3)));
    }

    public void newItems() {
        if (MainActivity.shownDialogBuyFromMerchant != null) {
            MainActivity.shownDialogBuyFromMerchant.dismiss();
        }
        initialize(null);
    }

    @Override // androidx.fragment.app.DialogFragment, androidx.fragment.app.Fragment
    public void onStart() {
        super.onStart();
        MainActivity.shownDialogMerchant = this;
    }

    @Override // androidx.fragment.app.DialogFragment, androidx.fragment.app.Fragment
    public void onStop() {
        MainActivity.shownDialogMerchant = null;
        super.onStop();
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/ui/dialogs/DialogMerchant.kt
package it.paranoidsquirrels.idleguildmaster.ui.dialogs

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ListAdapter
import androidx.viewbinding.ViewBinding
import it.paranoidsquirrels.idleguildmaster.MainActivity
import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.TrueTimeUtils
import it.paranoidsquirrels.idleguildmaster.UIUtils
import it.paranoidsquirrels.idleguildmaster.Utils
import it.paranoidsquirrels.idleguildmaster.databinding.DialogMerchantBinding
import it.paranoidsquirrels.idleguildmaster.storage.data.items.Item
import it.paranoidsquirrels.idleguildmaster.storage.data.items.MerchantOffer
import it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Upgrade
import java.util.ArrayList
import java.util.function.BooleanSupplier

class DialogMerchant : CustomDialog() {
    @JvmField
    var binding: DialogMerchantBinding? = null

    override fun getBinding(): ViewBinding = binding!!

    override fun setBinding(viewBinding: ViewBinding) {
        binding = viewBinding as DialogMerchantBinding
    }

    override fun getTitle(): String = getString(R.string.merchant_dialog_title)

    override fun inflate(layoutInflater: LayoutInflater, viewGroup: ViewGroup?, attachToRoot: Boolean): ViewBinding {
        val b = DialogMerchantBinding.inflate(layoutInflater, viewGroup, attachToRoot)
        binding = b
        return b
    }

    override fun initialize(arguments: Bundle?) {
        val b = binding ?: return
        b.newRegular.visibility = if (MainActivity.data.isNewMerchantRegularItems) 0 else 4
        b.newSpecial.visibility = if (MainActivity.data.isNewMerchantSpecialItems) 0 else 4
        MainActivity.data.isNewMerchantRegularItems = false
        MainActivity.data.isNewMerchantSpecialItems = false
        (activity as? MainActivity)?.refreshIcons()

        val regularItems = ArrayList<Item>()
        for (offer in MainActivity.data.merchantRegularStockItems) {
            offer.item?.let { regularItems.add(it) }
        }
        b.regularItemGrid.adapter = UIUtils.getItemsGridAdapter(context, regularItems) as ListAdapter

        val specialItems = ArrayList<Item>()
        for (offer in MainActivity.data.merchantSpecialReserve) {
            offer.item?.let { specialItems.add(it) }
        }
        b.specialItemGrid.adapter = UIUtils.getItemsGridAdapter(context, specialItems) as ListAdapter

        val isRegularEmpty = MainActivity.data.merchantRegularStockItems.isEmpty()
        b.regularItemGrid.visibility = if (isRegularEmpty) 8 else 0
        b.noRegularItems.visibility = if (isRegularEmpty) 0 else 8

        val isSpecialEmpty = MainActivity.data.merchantSpecialReserve.isEmpty()
        b.specialItemGrid.visibility = if (isSpecialEmpty) 8 else 0
        b.noSpecialItems.visibility = if (isSpecialEmpty) 0 else 8

        Utils.refreshCooldowns(TrueTimeUtils.millis())
    }

    override fun attachListeners() {
        val b = binding ?: return
        b.regularItemGrid.setOnItemClickListener { _, _, position, _ ->
            openBuyDialog(MainActivity.data.merchantRegularStockItems[position], false)
        }
        b.specialItemGrid.setOnItemClickListener { _, _, position, _ ->
            openBuyDialog(MainActivity.data.merchantSpecialReserve[position], true)
        }
        b.regularItemGrid.setOnItemLongClickListener { _, _, position, _ ->
            MainActivity.data.merchantRegularStockItems[position].item?.let { item ->
                UIUtils.vibrate(context)
                UIUtils.openItemDetail(item)
            }
            true
        }
        b.specialItemGrid.setOnItemLongClickListener { _, _, position, _ ->
            MainActivity.data.merchantSpecialReserve[position].item?.let { item ->
                UIUtils.vibrate(context)
                UIUtils.openItemDetail(item)
            }
            true
        }
        b.close.setOnClickListener {
            dismiss()
        }
    }

    private fun openBuyDialog(merchantOffer: MerchantOffer, isSpecial: Boolean) {
        if (MainActivity.shownDialogBuyFromMerchant != null) {
            return
        }
        val dialog = DialogBuyFromMerchant()
        dialog.offer = merchantOffer
        dialog.callback = BooleanSupplier {
            handleBuy(merchantOffer, isSpecial)
        }
        dialog.show(parentFragmentManager, "select_equipment")
    }

    private fun handleBuy(merchantOffer: MerchantOffer, isSpecial: Boolean): Boolean {
        val offerItem = merchantOffer.item ?: return false
        if (offerItem !is Upgrade && Utils.remainingInventorySpaceAfterCollecting(false, offerItem) < 0) {
            MainActivity.shownDialogBuyFromMerchant?.writeError(getString(R.string.error_not_enough_space))
            return false
        }
        if (merchantOffer.isGems) {
            val gems = MainActivity.data.gems - merchantOffer.price
            if (gems < 0) {
                MainActivity.shownDialogBuyFromMerchant?.writeError(getString(R.string.error_not_enough_gems))
                return false
            }
            MainActivity.data.gems = gems
            (activity as? MainActivity)?.refreshGems()
        } else {
            val money = MainActivity.data.money - merchantOffer.price
            if (money < 0) {
                MainActivity.shownDialogBuyFromMerchant?.writeError(getString(R.string.error_not_enough_money))
                return false
            }
            MainActivity.data.money = money
            (activity as? MainActivity)?.refreshMoney()
        }

        if (isSpecial) {
            MainActivity.data.merchantSpecialReserve.remove(merchantOffer)
        } else {
            MainActivity.data.merchantRegularStockItems.remove(merchantOffer)
        }

        if (offerItem.getUniqueOrigin() != null) {
            MainActivity.data.uniqueItemsLost.remove(offerItem.getUniqueOrigin())
        }

        if (offerItem is Upgrade) {
            (offerItem as Upgrade).use()
        } else {
            Utils.collectItem(offerItem, MainActivity.data.items)
        }

        MainActivity.headquartersFragment?.refresh()
        initialize(null)
        return true
    }

    fun refreshCooldowns(days: Int, hours: Int, minutes: Int) {
        binding?.regularItemsCountdown?.text = String.format(getString(R.string.time_hours_minutes), hours, minutes)
        binding?.specialItemsCountdown?.text = String.format(getString(R.string.time_days_hours_minutes), days, hours, minutes)
    }

    fun newItems() {
        MainActivity.shownDialogBuyFromMerchant?.dismiss()
        initialize(null)
    }

    override fun onStart() {
        super.onStart()
        MainActivity.shownDialogMerchant = this
    }

    override fun onStop() {
        MainActivity.shownDialogMerchant = null
        super.onStop()
    }
}
